metron-deployment/scripts/platform-info.sh

```diff
@@ -73,6 +73,30 @@ echo "--"
 echo "npm"
 npm --version
 
+# C++ compiler
+echo "--"
+echo "g++"
+g++ --version
+
+# C++11 compliant compiler
+echo "--"
+OBJFILE=/tmp/test
+CPPFILE=/tmp/test.cpp
+cat > $CPPFILE <<- EOM
+#include <iostream>
+using namespace std;
+int main() {
+    cout << "Hello World!" << endl;
+    return 0;
+}
+EOM
+g++ -std=c++11 $CPPFILE -o $OBJFILE
+if [ $? -eq 0 ]; then
+    echo "Compiler is C++11 compliant"
+else
+    echo "Warning: Compiler is NOT C++11 compliant"
+fi
+rm -f $CPPFILE $OBJFILE
 
```

Comment:
A standard way in the language is to check for the value of __cplusplus

201402L  is CPP 14
201103L is CPP 11

I don't have a preference, just throwing this out there.

Reply:
Do we have to account for platform_info on a non-dev machine?  If there is no g++ at all?

Reply:
> Do we have to account for platform_info on a non-dev machine? If there is no g++ at all?

The platform-info script is there to tell us whether a machine can be used to deploy metron.  Right now, you have to build Metron to deploy it, so the script checks all dev dependencies.

If g++ is not in the path, then the script will tell us that.  And it would carry on its merry way.  That's the behavior that I would expect.

Maybe I am not following you. Can you describe the scenario you are thinking of more?

Reply:
> A standard way in the language is to check for the value of __cplusplus

Thanks, I saw that.  I didn't think it was worth the trouble.  IMHO

Reply:
i agree about it not being worth the trouble.

as far as g++, I would expect us to check (pseudo)

if c++ in path then
    build test program
else
    echo No Cpp, you are in trouble!
fi

Maybe I'm mis-reading this, if there is no cpp, it will still try to build etc won't it?

Reply:
OK, I ran it, I did mis-understand.  sorry.

Reply:
No worries.  Glad to have a second set of eyes on it.